es-vue-base/src/lib-components/EsFileUpload.vue

```diff
@@ -0,0 +1,230 @@
+            active: false,
+        };
+    },
+    computed: {
+        fileTypesAsString() {
+            return this.fileTypes.join(', ');
+        },
+    },
+    watch: {
+        uploadUrls: {
+            handler(newUrls) {
+                if (newUrls.length > 0 && newUrls.length === this.files.length) this.upload();
+            },
+            deep: true,
+        },
+        files: {
+            handler(newFiles) {
+                if (newFiles.length > 0 && newFiles.length === this.uploadUrls.length) this.upload();
+            },
+            deep: true,
+        },
+        pickedItems(newVal) {
+            this.files = newVal;
+            if (newVal.length > 0 && this.fileSizeValid()) {
+                this.$emit('readyToUpload', this.files.length);
+                this.readFilesIntoUrl(newVal);
+            }
+        },
+    },
+    methods: {
+        fileSizeValid() {
+            // Check if the combined file sizes exceed the maxFileSize prop
+            const totalFileSize = this.files.reduce((acc, file) => acc + file.size, 0);
+            if (totalFileSize > this.maxFileSize * 1000000) {
+                this.$emit('fileSizeError');
+                return false;
+            }
+            return true;
+        },
+        readFilesIntoUrl(files) {
+            this.fileUrls = [];
+            files
+                .forEach((file) => {
+                    const fileReader = new FileReader();
+                    fileReader.onload = () => {
+                        this.$emit('fileDataRead', {
+                            name: file.name, type: file.type, size: file.size, data: fileReader.result,
+                        });
+                    };
+                    fileReader.readAsDataURL(file);
+                });
+        },
+        onDrop(event) {
+            // The user has dropped files onto the component. We have to apply the same logic as if they had
+            // selected the files from the file picker which limits the file types to the ones specified in the
+            // fileTypes prop.
+            this.active = false;
+            this.files = [];
+
+            // Use DataTransferItemList interface to access the file(s)
+            const invalidFiles = [...event.dataTransfer.items].filter((item) => item.kind !== 'file');
+
+            const dataTransfersAsFiles = [...event.dataTransfer.items]
+                .filter((item) => item.kind === 'file')
+                .map((item) => item.getAsFile());
+
+            const validFiles = dataTransfersAsFiles
+                .filter((file) => this.fileTypes.includes(file.type));
+            const fileErrors = dataTransfersAsFiles
+                .filter((file) => !this.fileTypes.includes(file.type))
+                .concat(invalidFiles)
+                .map((item) => item.name);
+            if (fileErrors.length > 0) {
+                this.$emit('fileTypeError', fileErrors);
+            }
```

Comment:
Although I like the emit, showing this error should be the responsibility of this component.